operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/DefaultContext.java

```diff
@@ -4,21 +4,51 @@
 import io.javaoperatorsdk.operator.processing.event.EventList;
 import java.util.Optional;
 
+/**
+ * Default implementation of the {@link Context} interface to be used by users when no custom
+ * implementation of that interface is needed.
+ *
+ * @param <T> the {@link CustomResource} the context is built around
+ */
 public class DefaultContext<T extends CustomResource> implements Context<T> {
 
+  /**
+   * Object containing information on how retries are processed.
+   */
   private final RetryInfo retryInfo;
+
+  /**
+   * List of events relating to the custom resource in context.
+   */
   private final EventList events;
 
+  /**
+   * Provides an instance given a list of events and the object encapsulating information about
+   * retries being currently processed, both relating to the given custom resource.
+   *
+   * @param events the {@link EventList} containing the relevant events to the resource
+   * @param retryInfo the {@link RetryInfo} object encapsulating info about retries
+   */
   public DefaultContext(EventList events, RetryInfo retryInfo) {
     this.retryInfo = retryInfo;
     this.events = events;
   }
 
+  /**
+   * Gets the events associated with this context instance.
```

Comment:
We shouldn't be documenting implementations of interface / abstract class methods unless their implementation needs to the be specifically documented and/or we want to call out the users' attention on a particular detail.